#### Iniciando processo de tratamento de dados. É preciso tratar os dados para posteriormente fazer uma análise exploratória.

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

dataframe = pd.read_csv(r'../data/raw/ipca.csv', index_col='Unnamed: 0')
pd.options.display.max_columns = 14
pd.options.display.max_rows = 20
pd.options.display.max_colwidth = 80
dataframe.info()
df = dataframe.copy()

<class 'pandas.core.frame.DataFrame'>
Index: 467 entries, 0 to 466
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   data    467 non-null    object 
 1   valor   467 non-null    float64
dtypes: float64(1), object(1)
memory usage: 10.9+ KB


In [2]:
df.shape

(467, 2)

Existem 2 colunas e 466 registros

Verificarei coluna por coluna, primeiramente vamos para coluna de data


requisitos: transformar em datetime e manter a estrutura padrão dd/mm/yyyy

In [3]:
rename_columns = {'data': 'periodo', 'valor': 'valor'}
df =  df.rename(columns=rename_columns)

In [4]:
df['periodo'] = df['periodo'].astype(str)
df['periodo'] = pd.to_datetime(df['periodo'], dayfirst=True)
#transformando df['periodo'] em datetime
df

periodo  valor
0   1986-08-01   2.57
1   1986-09-01   2.94
2   1986-10-01   1.96
3   1986-11-01   2.37
4   1986-12-01   5.47
..         ...    ...
462 2025-02-01   0.99
463 2025-03-01   0.96
464 2025-04-01   1.06
465 2025-05-01   1.14
466 2025-06-01   0.54

[467 rows x 2 columns]

Verificar valores nulos

In [5]:
df['periodo'].isnull().value_counts()
# há 467 registros não nulos

periodo
False    467
Name: count, dtype: int64

In [6]:
df['periodo'].unique()

<DatetimeArray>
['1986-08-01 00:00:00', '1986-09-01 00:00:00', '1986-10-01 00:00:00',
 '1986-11-01 00:00:00', '1986-12-01 00:00:00', '1987-01-01 00:00:00',
 '1987-02-01 00:00:00', '1987-03-01 00:00:00', '1987-04-01 00:00:00',
 '1987-05-01 00:00:00',
 ...
 '2024-09-01 00:00:00', '2024-10-01 00:00:00', '2024-11-01 00:00:00',
 '2024-12-01 00:00:00', '2025-01-01 00:00:00', '2025-02-01 00:00:00',
 '2025-03-01 00:00:00', '2025-04-01 00:00:00', '2025-05-01 00:00:00',
 '2025-06-01 00:00:00']
Length: 467, dtype: datetime64[ns]

Verificando duplicados

In [7]:
df['periodo'].duplicated().value_counts()
# há 467 registros não duplicados

periodo
False    467
Name: count, dtype: int64

Transformar a coluna de valor em float, verificar valores nulos

In [8]:
df['valor'].astype(float)

0      2.57
1      2.94
2      1.96
3      2.37
4      5.47
       ... 
462    0.99
463    0.96
464    1.06
465    1.14
466    0.54
Name: valor, Length: 467, dtype: float64

In [9]:
df['valor'].unique()

array([ 2.57,  2.94,  1.96,  2.37,  5.47, 11.  , 19.61, 11.95, 15.3 ,
       24.63, 18.02,  8.91,  8.09,  7.99,  9.45, 12.92, 14.38, 16.78,
       18.35, 16.59, 20.25, 18.65, 20.17, 24.69, 22.63, 26.25, 29.79,
       28.41, 30.24, 22.97, 18.95, 20.41, 11.52, 11.43, 27.29, 33.15,
       35.49, 38.58, 47.7 , 48.41, 64.21, 67.6 , 82.04, 36.76,  4.23,
        5.69,  8.73, 13.79, 11.53, 15.21, 16.49, 19.83, 22.86, 21.02,
        6.85,  8.99,  9.67,  9.56, 10.32, 12.39, 15.75, 19.78, 25.95,
       32.43, 31.17, 29.06, 28.76, 26.86, 23.92, 23.  , 24.28, 26.21,
       25.65, 27.66, 28.18, 26.4 , 25.92, 28.52, 28.9 , 28.36, 30.53,
       30.9 , 31.91, 32.73, 34.64, 37.23, 38.4 , 38.38, 40.38, 42.76,
       41.99, 46.42, 46.49, 47.95, 50.62,  6.87,  4.17,  3.83,  3.62,
        4.07,  3.8 ,  3.37,  3.25,  4.26,  4.25,  4.04,  4.02,  3.84,
        3.32,  3.09,  2.88,  2.78,  2.58,  2.35,  2.22,  2.07,  2.01,
        1.98,  1.93,  1.97,  1.9 ,  1.86,  1.8 ,  1.73,  1.67,  1.64,
        1.66,  1.58,

In [10]:
df['valor'].isnull().value_counts()
# há 467 valores não nulos

valor
False    467
Name: count, dtype: int64

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 467 entries, 0 to 466
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   periodo  467 non-null    datetime64[ns]
 1   valor    467 non-null    float64       
dtypes: datetime64[ns](1), float64(1)
memory usage: 10.9 KB


In [12]:
caminho = 'data/processed'
df.to_csv(f'../{caminho}/ipca.csv')